homeassistant/components/solarlog/sensor.py

```diff
@@ -250,26 +267,24 @@ async def async_setup_entry(
 class SolarLogCoordinatorSensor(SolarLogCoordinatorEntity, SensorEntity):
     """Represents a SolarLog sensor."""
 
-    entity_description: SolarLogSensorEntityDescription
+    entity_description: SolarLogCoordinatorSensorEntityDescription
 
     @property
-    def native_value(self) -> float | datetime:
+    def native_value(self) -> StateType | datetime | None:
         """Return the state for this sensor."""
 
-        val = self.coordinator.data[self.entity_description.key]
-        return self.entity_description.value_fn(val)
+        return self.entity_description.value_fn(self.coordinator.data)
 
 
 class SolarLogInverterSensor(SolarLogInverterEntity, SensorEntity):
     """Represents a SolarLog inverter sensor."""
 
-    entity_description: SolarLogSensorEntityDescription
+    entity_description: SolarLogInverterSensorEntityDescription
 
     @property
-    def native_value(self) -> float | datetime:
+    def native_value(self) -> float | None:
```

Comment:
```suggestion
    def native_value(self) -> StateType:
```

Reply:
Ok for me. I did not use StateType as (at least for the time being) only float and None are used...